application.go

```diff
@@ -609,6 +628,16 @@ func importApplication(fields schema.Fields, defaults schema.Defaults, importVer
 	if importVersion >= 4 {
 		result.Placement_ = valid["placement"].(string)
 		result.DesiredScale_ = int(valid["desired-scale"].(int64))
+
+		if result.Type_ == CAAS {
```

Comment:
I don't think we want business-level checks like this here - if we get a serialised application that's the right version with operator status set, we want to deserialise it.

Hmm, I can see that there are lots of places that are doing that already :(, but I think they're wrong, so let's not add to it here.